search/search.go

```diff
@@ -194,21 +233,33 @@ func (x *Index) Get(c context.Context, id string, dst interface{}) error {
 
 // Delete deletes a document from the index.
 func (x *Index) Delete(c context.Context, id string) error {
+	return x.DeleteMulti(c, []string{id})
+}
+
+// DeleteMulti deletes multiple documents from the index.
+func (x *Index) DeleteMulti(c context.Context, ids []string) error {
+	if len(ids) > maxDocumentsPerPutDelete {
+		return ErrTooManyDocuments
+	}
+
 	req := &pb.DeleteDocumentRequest{
 		Params: &pb.DeleteDocumentParams{
-			DocId:     []string{id},
+			DocId:     ids,
 			IndexSpec: &x.spec,
 		},
 	}
 	res := &pb.DeleteDocumentResponse{}
 	if err := internal.Call(c, "search", "DeleteDocument", req, res); err != nil {
 		return err
 	}
-	if len(res.Status) != 1 {
-		return fmt.Errorf("search: internal error: wrong number of results (%d)", len(res.Status))
+	if len(res.Status) != len(ids) {
+		return fmt.Errorf("search: internal error: wrong number of results (%d, expected %d)",
+			len(res.Status), len(ids))
 	}
-	if s := res.Status[0]; s.GetCode() != pb.SearchServiceError_OK {
-		return fmt.Errorf("search: %s: %s", s.GetCode(), s.GetErrorDetail())
+	for _, s := range res.Status {
+		if s.GetCode() != pb.SearchServiceError_OK {
+			return fmt.Errorf("search: %s: %s", s.GetCode(), s.GetErrorDetail())
```

Comment:
Will this ever be a partial delete (some success, some fail)? If so, this error could be misleading. Should we aggregate all the errors? Or else indicate in the error message that some documents may have been deleted?